scripts/globals/helm.lua

```diff
@@ -1535,6 +1535,20 @@ xi.helm.onTrade = function(player, npc, trade, helmType, csid, func)
                 movePoint(player, npc, zoneId, info)
             end
 
+            if
+                xi.events and
+                xi.events.egg_hunt and
+                xi.events.egg_hunt.enabledCheck and
+                player:getVar("[EGGHUNT]DAILY_HELM") < vanaDay()
```

Comment:
Dunno if they ever changed it (was kind of annoying and I thought it was an oversight) but used to be able to cheese the event by re-zoning because every 1st helm attempt would generate another egg even tho the day had not changed. its worth making a note to confirm next year (no change requested here!)

Reply:
Yeah this implementation definitely won't do that, hopefully that's not a thing on retail or has been patched out! If it is, switching this to a local var should be an easy fix, or maybe make it an optional setting in future.

Reply:
Right, eggxactly